packages/superset-ui-chart/src/query/processFilters.ts

```diff
@@ -0,0 +1,51 @@
+import { ChartFormData } from '../types/ChartFormData';
+import { QueryObjectFilterClause } from '../types/Query';
+import { isSimpleAdhocFilter } from '../types/Filter';
+import convertFilter from './convertFilter';
+
+/** Logic formerly in viz.py's process_query_filters */
+export default function processFilters(formData: ChartFormData) {
+  // TODO: Implement
+  // utils.convert_legacy_filters_into_adhoc(self.form_data)
+
+  // TODO: Implement
+  // merge_extra_filters(self.form_data)
+
+  // utils.split_adhoc_filters_into_base_filters(self.form_data)
+  // eslint-disable-next-line camelcase
+  const { adhoc_filters } = formData;
+  if (Array.isArray(adhoc_filters)) {
+    const simpleWhere: QueryObjectFilterClause[] = [];
```

Comment:
(not blocking) the type helps but I find these a little confusing name-wise, not sure what to expect `simple` to mean, maybe `whereFilter` <> `whereFreeform` and `havingFilter` <> `havingFreeform`?